### PR TITLE
feat(#2327): remove the old hash attribute

### PR DESCRIPTION
**Pull request by volodya-lombrozo**
Remove the old `hash` attribute from `AssembleMojo` and rename new `hsh` attribute to `hash`.

Closes: #2327

---

### Detailed summary
- Renamed `hsh` parameter to `hash` in `PullMojo` and `ProbeMojo` classes.
- Updated references to the renamed parameter in affected classes.
- Removed unused `hash` parameter from `pom.xml` files.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: eo-maven-plugin/src/main/java/org/eolang/maven/AssembleMojo.java [hunk 1]
```diff
@@ -89,19 +89,6 @@ public final class AssembleMojo extends SafeMojo {
...
-     * @since 0.21.0
-     * @todo #2302:30min Rename the parameter "hash". This parameter is actually
-     *  a tag, not a hash. By this tag application actually finds hash and then
-     *  uses it. So need to rename this parameter to "tag" and rename it in all
-     *  places where it's used. Also it would be better to use name "hash" for
-     *  parameter "hsh" in {@link ProbeMojo} and {@link PullMojo}
-     */
-    @SuppressWarnings("PMD.ImmutableField")
-    @Parameter(property = "eo.hash", required = true, defaultValue = "master")
-    private String hash = "master";
```
Comment (Graur) [inline eo-maven-plugin/src/main/java/org/eolang/maven/AssembleMojo.java:103]: @volodya-lombrozo As far as I understand this issue was about to rename `hash` parameter to `tag`. Could you please help me to understand why did you remove it at all?

**Review by Graur — APPROVED** [review]
@volodya-lombrozo Good one!
@yegor256 Please have a look

**Comment by volodya-lombrozo** [comment]
@yegor256 Could you merge that changes, please?

**Comment by volodya-lombrozo** [comment]
@yegor256 Reminder

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**